src/lambda_handler.py

```diff
@@ -36,54 +203,257 @@ def task_default(event):
+        "file_header",
+        [
+            "###",
+            "# https://{domain}/{filename}",
+            "# {timestamp}",
+            "# {description}",
+            "###",
+        ],
+    )
+
+    # AWS resource tag name indicating whether an IP address should be published
+    publish_egress_tag_name: str = validated_event.get(
+        "publish_egress_tag", "Publish Egress"
+    )
+
+    # An AWS-style filter definition to limit the queried regions
+    region_filters: List[Dict[str, Union[str, List[str]]]] = validated_event.get(
+        "region_filters", []
+    )
+
+    for account_id in account_ids:
+        logging.info("Examining account: %s", account_id)
+
+        # Create an EC2 client with the assumed role
+        ec2: boto3.client = create_assumed_aws_client(
+            aws_service="ec2",
+            role_arn=f"arn:aws:iam::{account_id}:role/{ec2_read_role_name}",
+            session_name="publish-egress-ip-lambda",
+        )
+
+        # Get a list of all regions that match our filter
+        regions: List[str] = get_ec2_regions(ec2, region_filters)
+
+        logging.info("Gathering public IPs from %d regions", len(regions))
+
+        # Loop through the region list and fetch the public EC2 IPs
+        for region in regions:
+            logging.info("Querying region: %s", region)
+
+            # Create an EC2 resource with the assumed role in the specified
+            # region
+            ec2 = create_assumed_aws_resource(
+                aws_service="ec2",
+                region=region,
+                role_arn=f"arn:aws:iam::{account_id}:role/{ec2_read_role_name}",
+                session_name="publish-egress-ip-lambda",
+            )
+
+            # Get the public IPs of instances that are tagged to be published
+            for ec2_info in get_ec2_ips(
+                ec2, application_tag_name, publish_egress_tag_name
+            ):
+                # Loop through all regexes and add IP to set if matched
+                for config in file_configs:
+                    if config["app_regex"].match(ec2_info.application_tag_value):
+                        config["ip_set"].add(ip_network(ec2_info.public_ip))
+
+    # Use a single timestamp for all files
+    now = "{:%a %b %d %H:%M:%S UTC %Y}".format(datetime.utcnow())
+
+    # The domain to display in the header of each published file
+    domain: str = validated_event.get("domain", "example.gov")
+
+    # Update each object (file) in the bucket
+    for config in file_configs:
+        # Initialize contents of object to be published
+        object_contents = "\n".join(file_header) + "\n"
+        for net in collapse_addresses(config["ip_set"]):
+            object_contents += str(net) + "\n"
+
+        # Fill in header template
+        object_contents = object_contents.format(
+            domain=domain,
+            filename=config["filename"],
+            timestamp=now,
+            description=config["description"],
```

Comment:
Since these are kwargs I think it would be useful to alphabetize them.

```suggestion
            description=config["description"],
            domain=domain,
            filename=config["filename"],
            timestamp=now,
```